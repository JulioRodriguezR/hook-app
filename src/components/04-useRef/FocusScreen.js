import React, {useRef} from 'react'

export const FocusScreen = () => {

  // mutable Ref
  const inputRef = useRef();

  const handleClick = () => {
    inputRef.current.select();
    console.log(inputRef);
  }

  return (
    <div>
     <input ref={inputRef} className="form-control" placeholder="Your name"/>
     <button className="btn btn-outline-primary mt-5" onClick={handleClick}>Focus</button>
    </div>
  )
}
