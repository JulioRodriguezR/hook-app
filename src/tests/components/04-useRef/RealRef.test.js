import React from 'react';
import { shallow } from "enzyme";

import { RealRef } from '../../../components/04-useRef/RealRef';

describe('Test in <RealRef />', () => {

  test('Should show correctly', () => {
    const wrapper = shallow(<RealRef />);
    
    expect(wrapper).toMatchSnapshot(); 
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false); 
  });

  test('Should show <MultipleCustomHooks /> component', () => {
    const wrapper = shallow(<RealRef />);

    wrapper.find('button').simulate('click');

    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true); 
  });

});