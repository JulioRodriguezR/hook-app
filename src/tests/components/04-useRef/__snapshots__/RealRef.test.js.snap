// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <RealRef /> Should show <MultipleCustomHooks /> component 1`] = `
<div>
  <h1>
    Real Example Ref Hook
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;

exports[`Test in <RealRef /> Should show correctly 1`] = `
<div>
  <h1>
    Real Example Ref Hook
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
